### DISCUSSION TITLE
Source Diff between two builds

### DISCUSSION THREAD

**Discussion by Nantha96** [concourse/concourse#7514] [Help & Support] [answered]
I just want to Compare the source between two builds. In my case Git Hub is the Source Repository. If my Build 1 deployed in XX Environment. Then my next Build is going to deploy the changes in XX Environment. Before deploy I need to compare the sources. Is there any way I can compare the sources. Like get the git commit value from the Build 1 or Store the Build 1 git commit in Pipeline variable.
Any suggestions or Ideas would be a great help for me.

**Comment by taylorsilva** [comment, marked as answer]
First thought I have is to create a [task](https://concourse-ci.org/tasks.html) that does something like `git diff HEAD~1`. I guess the hard part for you is actually getting the commit that last deployed. Maybe you could store the sha of the commit that last went to production somewhere like an s3 bucket?

**Reply by Nantha96** [reply]
Thanks, I too thought about the same thing to store the sha of the commit externally. Just wanted to make sure if something already existed in the concourse itself ..
